Source/WebGPU/WGSL/TypeCheck.cpp

```diff
@@ -112,6 +116,34 @@ TypeChecker::TypeChecker(ShaderModule& shaderModule)
     introduceVariable(AST::Identifier::make("i32"_s), m_types.i32Type());
     introduceVariable(AST::Identifier::make("u32"_s), m_types.u32Type());
     introduceVariable(AST::Identifier::make("f32"_s), m_types.f32Type());
+
+    // FIXME: Add all other overloads
+    // FIXME: we should make this a lot more convenient
+    // operator + [T<:Number](T, T) -> T
+    OverloadCandidate plus1;
+    {
+        TypeVariable T { 0, TypeVariable::Number };
+        plus1.typeVariables.append(T);
+        plus1.parameters.append(T);
+        plus1.parameters.append(T);
+        plus1.result = T;
```

Comment:
Is there a way to represent this as data rather than C++? I assume there are going to be dozens of more stanzas like this. Can we generate them at build time? (or run time?)

Reply:
Yeah, that's the plan. I'm hoping to do something similar to JSC's `BytecodeList.rb`, where we declare all these in a file and output these C++ definitions at compile time.